hw/ip/otbn/data/otbn_testplan.hjson

```diff
@@ -14,19 +13,46 @@
       desc: '''
             Basic sanity test acessing a major datapath within the otbn.
 
-            **Stimulus**:
-            - TBD
+            The so-called "sanity test" is actually the main bulk of OTBN
+            testing. This picks a random binary from a pre-generated set and
+            runs it, comparing against the model. We'll run this with a large
+            number of seeds and use functional coverage to track when
+            verification of the internals of the core is done.
+
+            Sometimes enable the "done" interrupt to check that it and the
+            error interrupt work correctly.
```

Comment:
not sure if we can make the sanity test simpler and more predictable, like just use a fixed binary which tests some basic function. And have another test to randomly pick any generated binary. So, when the sanity test fails, we just need to check very basic stuff as it's probably not due to some rare combination. Also, CI runs the sanity test and it intends to only check basic function.

Reply:
Agree with this.  Ideally the smoke test (I think that is the "new" terminology for this test) should quickly exercise most of the functions in the block.

For something like OTBN we could do some very small-size RSA - in the past I have seen something like RSA-128 :) which runs very fast but uses all the same instructions/functional units as 2k/3k/4k RSA - the test can do an encrypt followed by decrypt to self-check.  I could imagine randomizing the key and using fixed "hello world" type data blobs.

Reply:
Hmm, we could do that. Don't forget that we also have Greg's existing smoke test infrastructure (running in CI) to catch really silly breakages. Honestly, the RTL is already stable enough, and we have a separate CI smoke test, so I don't think this "super-simple" test would ever catch anything.

I guess I could split this into something like "smoke_test" and "single_binary", where "smoke_test" runs the code that's already checked in as a smoke test (and gets run already anyway), and "single_binary" is the actually useful test. But I'd definitely want to run the latter in CI as well, so I'm not really sure there's much point.

What do you think?

Reply:
Thinking about it further, I don't think that splitting out a "run a known binary" test is such a bad idea. I've pushed up a new version of the testplan that does so. Does that look reasonable, do you think?

Reply:
@cdgori: Philipp has done some work to get an RSA integration test working. I think that chip-level is probably a better place for it than here.